sycl/include/sycl/marray.hpp

```diff
@@ -216,6 +216,15 @@ template <typename Type, std::size_t NumElements> class marray {
   operator BINOP(const marray &Lhs, const T &Rhs) {                            \
     return Lhs BINOP marray(static_cast<DataT>(Rhs));                          \
   }                                                                            \
+  template <typename T>                                                        \
+  friend typename std::enable_if<                                              \
+      std::is_convertible<DataT, T>::value &&                                  \
+          (std::is_fundamental<T>::value ||                                    \
+           std::is_same<typename std::remove_const<T>::type, half>::value),    \
+      marray>::type                                                            \
```

Comment:
Are there any issues if using `std::is_convertible_v`/`std::is_same_v`/`std::enable_if_t`?

Reply:
It is just in line with the existing definitions. I've changed them for this and the other definitions to keep them consistent.